packages/eslint-plugin/src/rules/no-confusing-void-expression.ts

```diff
@@ -112,6 +139,20 @@ export default createRule<Options, MessageId>({
         if (invalidAncestor.type === AST_NODE_TYPES.ArrowFunctionExpression) {
           // handle arrow function shorthand
 
+          if (options.ignoreVoidInVoid) {
+            if (
+              (invalidAncestor.returnType &&
+                isVoidLikeType(invalidAncestor.returnType)) ||
```

Comment:
```suggestion
              invalidAncestor.returnType ||
```

All tests still pass without this `isVoidLikeType` call. We should either add some tests to prove that this line is really needed, or remove it.

Reply:
If a returnType exists, but the returnType is not void, it is not a valid typescript syntax, so there seems to be no problem with removing it.

```typescript
const arrowFn = ():undefined => console.log('foo'); // type error
```